server/src/main/java/org/apache/cassandra/sidecar/handlers/GossipUpdateHandler.java

```diff
@@ -0,0 +1,98 @@
+
+import java.util.Collections;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.Inject;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.net.SocketAddress;
+import io.vertx.ext.auth.authorization.Authorization;
+import io.vertx.ext.web.RoutingContext;
+import org.apache.cassandra.sidecar.acl.authorization.BasicPermissions;
+import org.apache.cassandra.sidecar.common.request.data.NodeCommandRequestPayload;
+import org.apache.cassandra.sidecar.common.server.StorageOperations;
+import org.apache.cassandra.sidecar.concurrent.ExecutorPools;
+import org.apache.cassandra.sidecar.utils.InstanceMetadataFetcher;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Handles {@code PUT /api/v1/cassandra/gossip} requests to start or stop Cassandra gossip.
+ *
+ * <p>Expects a JSON payload:
+ * { "state": "start" } or { "state": "stop" }
+ * and will asynchronously invoke the corresponding JMX operation.</p>
+ */
+public class GossipUpdateHandler extends NodeCommandHandler implements AccessProtected
+{
+    private final ExecutorPools executorPools;
+    private static final Logger LOGGER = LoggerFactory.getLogger(GossipUpdateHandler.class);
+
+
+    @Inject
+    public GossipUpdateHandler(InstanceMetadataFetcher metadataFetcher,
+                               ExecutorPools executorPools)
+    {
+        super(metadataFetcher, executorPools, null);
+        this.executorPools = executorPools;
+    }
+
+    @Override
+    public Set<Authorization> requiredAuthorizations()
+    {
+        return Collections.singleton(BasicPermissions.WRITE_GOSSIP.toAuthorization());
+    }
+
+    @Override
+    protected void handleInternal(RoutingContext context,
+                                  HttpServerRequest httpRequest,
+                                  @NotNull String host,
+                                  SocketAddress remoteAddress,
+                                  NodeCommandRequestPayload request)
+    {
+        StorageOperations storageOps = metadataFetcher.delegate(host).storageOperations();
+
+        executorPools.service()
+                     .runBlocking(() -> {
+                         switch (request.getState())
+                         {
+                             case START:
+                                 storageOps.startGossiping();
+                                 break;
+                             case STOP:
+                                 storageOps.stopGossiping();
+                                 break;
+                             default:
+                                 throw new IllegalStateException("Unknown state: " + request.getState());
+                         }
+                     })
+                     .onSuccess(ignored -> {
+                         context.response().setStatusCode(HttpResponseStatus.ACCEPTED.code()).end();
```

Comment:
we always respond with a JSON payload. Maybe status: OK , or status: Accepted are possible responses?